### PR TITLE
keep all custom Composer repositories

### LINKED ISSUE #382: Repository type is not registered - v1.0.81
The latest release seems to break [custom repositories](https://getcomposer.org/doc/05-repositories.md#vcs) feature in `composer.json`.

Step to reproduce:

Add this in your `composer.json`: 
```
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/0x46616c6b/monolog"
        }
    ],
```

Then run `composer update`.

Expected result: the update run properly

Actual result: 
```
Loading composer repositories with package information                                                                                                                         Updating dependencies (including require-dev)         
Package operations: 74 installs, 0 updates, 0 removals
  - Installing symfony/flex (v1.0.81): Loading from cache
Plugin installation failed, rolling back
  - Removing symfony/flex (v1.0.81)

                                          
  [InvalidArgumentException]              
  Repository type is not registered: vcs  
```

The update works fine if I manually set `symfony/flex` to v1.0.80.

composer v1.6.5
PHP 7.2.7

### PR DESCRIPTION
**Pull request by xabbuh**
This fixes #382.

### CODE REVIEW

File: src/Flex.php [hunk 1]
```diff
@@ -104,11 +103,13 @@ public function activate(Composer $composer, IOInterface $io)
 
         $rfs = Factory::createRemoteFilesystem($this->io, $this->config);
         $this->rfs = new ParallelDownloader($this->io, $this->config, $rfs->getOptions(), $rfs->isTlsDisabled());
+        $repositoryClassesProperty = new \ReflectionProperty(RepositoryManager::class, 'repositoryClasses');
+        $repositoryClassesProperty->setAccessible(true);
         $manager = new RepositoryManager($this->io, $this->config, $composer->getEventDispatcher(), $this->rfs);
```
Comment (xabbuh) [inline src/Flex.php]: An alternative solution would be to use `RepositoryFactory::manager()` here to create the `RepositoryManager` instance, but that would not take into account any changes that could have been made (by plugins, for example) to the original manager.

**Comment by stof** [comment]
isn't this missing adding the repos ?

**Comment by xabbuh** [comment]
Composer seems to do that itself by calling `createRepository()` based on the configured repository classes. At least, it did that on a local project where I have a `path` repository.

**Comment by xabbuh** [comment]
But we may indeed pull the formerly set repositories with `getRepositories()` and set them as is (except for the `composer` repository). What do you think?

**Pull request closed without merging**